core/src/main/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponse.java

```diff
@@ -56,6 +60,15 @@ public RoutingExplanations getExplanations() {
         return this.explanations;
     }
 
+    public List<String> getMessages() {
+        return explanations.explanations().stream()
+            .filter(explanation -> explanation.decisions().type().equals(Decision.Type.YES))
+            .map(explanation -> explanation.command().getMessage())
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.toList());
+    }
+
```

Comment:
It may make more sense to put this method in RoutingExplanations itself since that's what it operates on. My thinking here was that the logic of "which messages do we actually surface" is closer to the response handling.